great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -474,7 +481,7 @@ def instance_id(self) -> str:
             if self._in_memory_instance_id is not None:
                 return self._in_memory_instance_id
             instance_id = str(uuid.uuid4())
-            self._in_memory_instance_id = instance_id
+            self._in_memory_instance_id = instance_id  # type: ignore[assignment]
```

Comment:
I think we can fix this by defining this field as `str | None` on line 269. I would do that rather than introduce a type ignore.

Side-rant: we should not be assigning anything in property, but let's leave that for another PR.

Reply:
Actually rethinking this... we can just do that fallback to the uuid within the constructor, right? I think we should do that and save ourselves a bunch of hoop jumping.